middleware/logging.go

```diff
@@ -39,7 +39,12 @@ func (l Log) Wrap(next http.Handler) http.Handler {
 		var buf bytes.Buffer
 		wrapped := newBadResponseLoggingWriter(w, &buf)
 		next.ServeHTTP(wrapped, r)
-		statusCode := wrapped.statusCode
+
+		statusCode, err := wrapped.statusCode, wrapped.writeError
+		if err != nil {
+			l.logWithRequest(r).Warnf("%s %s %s, error: %s ws: %v; %s", r.Method, uri, time.Since(begin), err, IsWSHandshakeRequest(r), headers)
```

Comment:
probably don't want a warning every time the user's client times out

Reply:
Agree. We could log with debug level if `grpc.IsCanceled()` and warning otherwise. Also, I think `grpc.IsCanceled()` is not fully correct because the error could be wrapped. We should improve it like this:

```
func IsCanceled(err error) bool {
	if errors.Is(err, context.Canceled) {
		return true
	}

```